eureka-client/src/main/java/com/netflix/eureka2/client/registration/RegistrationHandlerImpl.java

```diff
@@ -59,7 +59,7 @@ public RegistrationHandlerImpl(ClientChannelFactory channelFactory) {
             logger.info("Instance: {} is not registered. Ignoring unregister", instanceInfo);
             return Observable.empty(); // Be more acceptable to errors from user as unregister for non-existent instance is a no-op.
         }
-        return registrationChannel.unregister().doOnTerminate(new Action0() {
```

Comment:
What will happen now when there is an error during unregistration? Shouldn't we close the channel as well?